rskj-core/src/main/java/co/rsk/net/sync/ConnectionPointFinder.java

```diff
@@ -0,0 +1,52 @@
+package co.rsk.net.sync;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.util.Optional;
+
+/**
+ * Uses Binary Search to help find a connection point with another peer.
+ */
+public class ConnectionPointFinder {
+
+    // Status used to find connection point
+    private long start;
+    private long end;
+
+    // Connection point found or not
+    private Long connectionPoint = null;
+
+    public ConnectionPointFinder(long height) {
+        this.start = 0;
+        this.end = height;
+    }
+
+    public Optional<Long> getConnectionPoint() {
+        return Optional.ofNullable(this.connectionPoint);
+    }
+
+    public long getFindingHeight() {
+        return this.start + (this.end - this.start) / 2;
```

Comment:
(start + end) / 2

Reply:
This way helps to avoid overflow problems